src/marqo/tensor_search/validation.py

```diff
@@ -501,7 +510,66 @@ def validate_multimodal_combination(field_content, is_non_tensor_field, field_ma
             f"It must be a tensor field. Add this field to `tensor_fields` or"
             f"add it as a normal field to fix this problem."
         )
-    return True
+    return field_content
+
+
+def validate_custom_vector(field_content: dict, is_non_tensor_field: bool, index_model_dimensions: int):
+    '''
+    Validates the field content if it is a custom vector field (dict)
+    Args:
+        field_content: the field content
+        is_non_tensor_field: whether this is a non-tensor-field
+        index_model_dimensions: the `dimensions` property of the index to be added to
+
+    Returns:
+        field_content if the validation passes
+        "content" key will be added to the field_content as empty string if it is not provided.
+    '''
+    if index_model_dimensions is None:
```

Comment:
maybe check if the dimension is a `int`
```suggestion
if not isinstance(index_model_dimension, int): 
```